src/allowance/checkAllowance.ts

```diff
@@ -13,7 +14,8 @@ export const checkAllowance = async (
   statusManager: StatusManager,
   settings: InternalExecutionSettings,
   chain: Chain,
-  allowUserInteraction = false
+  allowUserInteraction = false,
+  configCallback?: (tx: TransactionRequest) => Promise<TransactionRequest>
```

Comment:
maybe approvalConfigCallback is a more descriptive name?

Reply:
I guess this is not only for approval but also for contract calls. https://lifi.atlassian.net/browse/LF-763

Maybe `updateTransactionRequest`?

Reply:
updated the variable name